### PR TITLE
add crc32c checksum type

### PR DESCRIPTION
**Pull request by junchaowu**
cc @breerly  @abhinav @blampe @jcorbin

### CODE REVIEW

**Comment by abhinav** [comment]
:shipit:

File: python/tchannel/messages/common.py [hunk 1]
```diff
@@ -85,6 +87,10 @@ def compute_checksum(checksum_type, args, csum=0):
     # TODO figure out farm32 cross platform issue
     elif checksum_type == ChecksumType.farm32:
         raise NotImplementedError()
+    elif checksum_type == ChecksumType.crc32c:
+        cfun = crcmod.predefined.mkCrcFun('crc-32c')
```
Comment (jcorbin) [inline python/tchannel/messages/common.py]: could this call into crcmod be done once at module time? or is cfun a stateful callable?

Comment (junchaowu) [inline python/tchannel/messages/common.py]: That will be better. Done.

**Comment by jcorbin** [comment]
lgtm

**Pull request merged**